public/main/client/index.js

```
@@ -106,26 +139,53 @@ function startCore({ chain, core, config: coreConfig }, webContent) {
-        await proxyRouterApi.kill(config.sellerProxyPort).catch(logger.error);
-        runProxyRouter(config, PROXY_ROUTER_MODE.Seller);
-      }
-
       const isBuyerHealthy = await isProxyRouterHealthy(
         api,
         config.localBuyerProxyRouterUrl
       );
-      if (!isBuyerHealthy) {
-        logger.debug("Buyer is not healhy, restart...");
-        await proxyRouterApi.kill(config.buyerProxyPort).catch(logger.error);
-        runProxyRouter(config, PROXY_ROUTER_MODE.Buyer);
+
+      if (
+        !isSellerHealthy ||
+        !isBuyerHealthy ||
+        restartDaemon ||
+        shouldRestartProxy
+      ) {
+        logger.debug("Proxy is not healhy, restart...");
+        if (os.platform() === "darwin") {
+          await proxyRouterApi.kill(config.sellerProxyPort).catch(logger.error);
+          await proxyRouterApi.kill(config.buyerProxyPort).catch(logger.error);
+          await runMacosDaemons(getResourcesPath(), config);
+        } else if (os.platform() === "win32") {
+          await proxyRouterApi.kill(config.sellerProxyPort).catch(logger.error);
+          await proxyRouterApi.kill(config.buyerProxyPort).catch(logger.error);
+          await runWindowsServices(getResourcesPath(), config);
+        } else if (os.platform() === "linux") {
+          await proxyRouterApi.kill(config.sellerProxyPort).catch(logger.error);
+          await proxyRouterApi.kill(config.buyerProxyPort).catch(logger.error);
+          await runLinuxDaemons(getResourcesPath(), config);
+        } else {
+          if (!isSellerHealthy) {
+            await proxyRouterApi
+              .kill(config.sellerProxyPort)
+              .catch(logger.error);
+            runProxyRouter(config, PROXY_ROUTER_MODE.Seller);
+          }
+
+          if (!isBuyerHealthy) {
+            await proxyRouterApi
+              .kill(config.buyerProxyPort)
+              .catch(logger.error);
+            runProxyRouter(config, PROXY_ROUTER_MODE.Buyer);
+          }
```

same here, it would look perfect if we could add one more "if" branch and on else throw error for unsupported platform

---

the idea is that we have a platform selector above this code, and here is "else" statement. So if platform is not supported daemons - we try to run proxy-router from node as we do now.